### PR TITLE
Change history of Spring Boot Starter Config

### LINKED ISSUE #10982: [Doc] Optimize change history of ShardingSphere-JDBC Spring Boot Starter Configuration (English)
Hi community,

This issue is for [#10887](https://github.com/apache/shardingsphere/issues/10887).

### Background

Refor to #10972

### Aim

Optimize change history of ShardingSphere-JDBC Spring Boot Starter Configuration. (English version). 

### How to

1. Migrate `Spring Boot Starter Configuration` change history from

   `4.1.2.Configuration Manual`->`Spring Boot Starter Configuration`->`Change History` to

   `7.2.API Change History`->`7.2.1.ShardingSphere-JDBC`->`Spring Boot Starter Configuration`;

2. Refer to the documents of each version on the official website, check whether the change history is correct, and fill in the missing parts.

3. The content organization of the change history is as follows:

   - ShardingSphere-5.0.0-beta
     - Sharding
     - Readwrite-splitting
     - Encryption
     - Shadow DB
     - Governance
   - ShardingSphere-4.x
     - Sharding
     - Readwrite-splitting
     - Data Masking
     - Orchestration
   - ShardingSphere-3.x
     - Sharding
     - Readwrite-splitting
     - Orchestration
   - ShardingSphere-2.x
     - Sharding
     - Readwrite-splitting
     - Orchestration

### Reference

- [Apache ShardingSphere website](https://shardingsphere.apache.org/)

- [Spring Boot Starter Configuration change history](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-jdbc/configuration/spring-boot-starter/)

### PR DESCRIPTION
**Pull request by Swastyy**
Fixes #10982 

Changes proposed in this pull request:
- Everything mentioned in the issue.

### CODE REVIEW

**Comment by tristaZero** [comment]
@Swastyy Hi Nice to see you again.

**Comment by tristaZero** [comment]
@RaigorJiang could you a have a look?

File: docs/document/content/others/api-change-history/shardingsphere-jdbc/spring-boot-starter.en.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title = "Change History"
 weight = 7
```
Comment (RaigorJiang) [inline docs/document/content/others/api-change-history/shardingsphere-jdbc/spring-boot-starter.en.md:3]: weight should be equal to '4', 
please refer to https://github.com/apache/shardingsphere/tree/master/docs/document/content/others/api-change-history/shardingsphere-jdbc

File: docs/document/content/others/api-change-history/shardingsphere-jdbc/spring-boot-starter.en.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ title = "Change History"
...
 +++
 
-## 5.0.0-alpha
+## 5.0.0-beta
 
 ### Replica Query
```
Comment (RaigorJiang) [inline docs/document/content/others/api-change-history/shardingsphere-jdbc/spring-boot-starter.en.md:8]: in 5.0.0, this feature named [READWRITE-SPLITTING](
https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-jdbc/configuration/spring-boot-starter/readwrite-splitting/), please change the name, thank you.

**Comment by RaigorJiang** [comment]
Hi @Swastyy ,
There are still some issues, let me list them together:
#### 5.0.0-beta
1. Readwrite-splitting
The content of this section is incorrect, such as
`spring.shardingsphere.rules.replica-query`, this key no longer exists, the chapter can be copied from [Readwrite-splitting](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-jdbc/configuration/spring-boot-starter/readwrite-splitting/).
2. Sharding
Can you put `sharding` in the first chapter? Because it is the most used feature of SS, like in other documents.

#### There is no problem with the content of 4.x and previous versions.
@Swastyy  thanks a lot!

**Comment by Swastyy** [comment]
Hi @RaigorJiang Please review it once.

**Comment by RaigorJiang** [comment]
> Hi @RaigorJiang Please review it once.

It's OK now, thank you 👍

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
@tristaZero Please merge.

**Review by tristaZero — APPROVED** [review]
Thanks, guys.

**Pull request merged**